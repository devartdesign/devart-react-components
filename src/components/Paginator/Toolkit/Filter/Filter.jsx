import React from 'react';
import PropTypes from 'prop-types';

import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';
import Checkbox from '@material-ui/core/Checkbox';

const MenuPropsStyle = {
  PaperProps: {
    style: {
      maxHeight: 48 * 4.5 + 8,
      width: 250,
    },
  },
};

export default class Filter extends React.PureComponent {
  static propTypes = {
    onSelect: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired,
    selectedOptions: PropTypes.array,
    label: PropTypes.string,
    name: PropTypes.string.isRequired,
    width: PropTypes.number
  };

  static defaultProps = {
    label: '',
    selectedOptions: [],
    width: 100
  }

  handleChange = (event) => {
    this.props.onSelect({
      name: event.target.name,
      value: event.target.value
    });
  };


  render() {
    const { name, options, label, selectedOptions, width } = this.props;
    return (
      <FormControl style={{ width }}>
        <InputLabel htmlFor="select-multiple-checkbox" style={{ top: -10, left: 4 }}>{label}</InputLabel>
        <Select
          name={name}
          multiple
          value={selectedOptions}
          onChange={this.handleChange}
          placeholder="Filters"
          input={<Input id="select-multiple-checkbox" />}
          renderValue={selected => selected.join(', ')}
          MenuProps={MenuPropsStyle}
          style={{ margin: 5 }}
        >
          {options.map(option => (
            <MenuItem key={`${label}-${option.key}`} value={option.key}>
              <Checkbox
                style={{ color: '#04b9e6' }}
                checked={selectedOptions.indexOf(option.key) > -1}
              />
              <ListItemText primary={option.label} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    );
  }
}
