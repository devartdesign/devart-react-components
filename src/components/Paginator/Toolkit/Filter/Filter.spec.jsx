import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Filter from './Filter';

describe('<PaginatorToolkitFilter />', () => {
  let wrapper;
  const defaultProps = {
    onSelect: jest.fn(),
    options: [{
      id: 'name',
      value: 'name',
      text: 'name',
    }, {
      id: 'status',
      value: 'status',
      text: 'status'
    }],
    selectedOptions: ['status'],
    label: 'Filter'
  };

  describe('render', () => {
    beforeEach(() => {
      wrapper = shallow(<Filter {...defaultProps} />);
    });

    it('should render <SearchIcon> and an <Input> components', () => {
      expect(renderer.create(wrapper)
        .toJSON())
        .toMatchSnapshot();
    });
  });

  describe('handleChange', () => {
    beforeEach(() => {
      wrapper = shallow(<Filter {...defaultProps} />);
      wrapper.instance().handleChange({
        target: {
          name: 'filter',
          value: ['name']
        }
      });
    });

    it('should call onSelect callback from Props', () => {
      expect(defaultProps.onSelect).toHaveBeenCalledWith({
        name: 'filter',
        value: ['name']
      });
    });
  });
});
