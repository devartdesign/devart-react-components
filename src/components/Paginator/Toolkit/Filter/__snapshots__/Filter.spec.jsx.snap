// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginatorToolkitFilter /> render should render <SearchIcon> and an <Input> components 1`] = `
<div
  className="MuiFormControl-root-1"
  style={
    Object {
      "width": 100,
    }
  }
>
  <label
    className="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
    data-shrink={true}
    htmlFor="select-multiple-checkbox"
    style={
      Object {
        "left": 4,
        "top": -10,
      }
    }
  >
    Filter
  </label>
  <div
    className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
    onClick={[Function]}
    style={
      Object {
        "margin": 5,
      }
    }
  >
    <div
      className="MuiSelect-root-23"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        status
      </div>
      <input
        id="select-multiple-checkbox"
        placeholder="Filters"
        type="hidden"
        value="status"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-60 MuiSelect-icon-29"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
